src/test/cypress/e2e/course/CourseCommunication.cy.ts

```diff
@@ -143,6 +203,64 @@ describe('Course communication', () => {
         });
     });
 
+    describe('Exercise communication', () => {
+        let textExercise: TextExercise;
+
+        before('Create exercise', () => {
+            cy.login(admin);
+            courseManagementRequest.createTextExercise({ course }).then((response) => {
+                textExercise = response.body;
+            });
+        });
+
+        it('students should be able to create posts within exercises', () => {
+            cy.login(studentOne, `/courses/${course.id}/exercises/${textExercise.id}`);
+            courseCommunication.newPost();
+            courseCommunication.setTitle('Exercise Test Post');
+            cy.fixture('loremIpsum.txt').then((text) => {
+                courseCommunication.setContent(text);
+            });
+            courseCommunication.save();
+        });
+
+        it('students should be able to search for exercise posts', () => {
+            const title = 'Exercise Filter Test Post';
+            const content = 'Exercise Filter Test Post Content';
+            cy.login(studentOne, `/courses/${course.id}/exercises/${textExercise.id}`);
+            courseManagementRequest.createCourseExercisePost(course, textExercise, title, content).then((response) => {
+                const post = response.body;
+                cy.login(studentTwo, `/courses/${course.id}/exercises/${textExercise.id}`);
+                cy.reload();
+                courseCommunication.searchForPost(title);
+                courseCommunication.checkSingleExercisePost(post.id, title, content);
+            });
+        });
+
+        it('other students should be able to reply to an exercise post', () => {
+            const title = 'My Reply Test Post';
+            const content = 'Test Reply Post Content';
+            cy.login(studentOne, `/courses/${course.id}/exercises/${textExercise.id}`);
+            courseManagementRequest.createCourseExercisePost(course, textExercise, title, content).then((response) => {
+                const post = response.body;
+                cy.login(studentTwo, `/courses/${course.id}/exercises/${textExercise.id}`);
+                cy.reload();
+                courseCommunication.reply(post.id, 'My Test reply');
+            });
+        });
+
+        it('other students should be able to react to exercise post', () => {
+            const title = 'My React Test Post';
+            const content = 'Test React Post Content';
+            cy.login(studentOne, `/courses/${course.id}/discussion`);
+            courseManagementRequest.createCourseExercisePost(course, textExercise, title, content).then((response) => {
+                const post = response.body;
+                cy.login(studentTwo, `/courses/${course.id}/exercises/${textExercise.id}`);
+                cy.reload();
+                courseCommunication.react(post.id, 'thumbs up');
```

Comment:
For ``reply`` and ``react``, as far as I see you only check if there's a button they can press to actually perform the action. Then you click it. However, shouldn't there be some sort of verification if the action has actually worked (e.g. reload, then check for the reaction)?

Reply:
Good point! I initially wanted to add that within a follow-up PR, but it makes more sense to include it in this one. I added the requested checks in the latest commit. Thanks for the hint!